examples/StandardFirmata/StandardFirmata.ino

```diff
@@ -108,6 +151,90 @@ byte wireRead(void)
  * FUNCTIONS
  *============================================================================*/
 
+// get a pointer to the serial port associated with the specified port id
+Stream* getPortFromId(byte portId)
+{
+  switch (portId) {
+    case HW_SERIAL0:
+      // block use of Serial (typically pins 0 and 1) until ability to reclaim Serial is implemented
+      //return &Serial;
+      return NULL;
+#if defined(UBRR1H) || defined(USBCON)
+    case HW_SERIAL1:
+      return &Serial1;
+#endif
+#if defined(UBRR2H) || defined(SERIAL_PORT_HARDWARE2)
+    case HW_SERIAL2:
+      return &Serial2;
+#endif
+#if defined(UBRR3H) || defined(SERIAL_PORT_HARDWARE3)
+    case HW_SERIAL3:
+      return &Serial3;
+#endif
+#if defined(ARDUINO_ARCH_AVR)
+    case SW_SERIAL0:
+      if (swSerial0 != NULL) {
+        // instances of SoftwareSerial are already pointers so simply return the instance
+        return swSerial0;
+      }
+      break;
+    case SW_SERIAL1:
+      if (swSerial1 != NULL) {
+        return swSerial1;
+      }
+      break;
+    case SW_SERIAL2:
+      if (swSerial2 != NULL) {
+        return swSerial2;
+      }
+      break;
+    case SW_SERIAL3:
+      if (swSerial3 != NULL) {
+        return swSerial3;
+      }
+      break;
+#endif
+  }
+  return NULL;
+}
+
+// Check serial ports that have READ_CONTINUOUS mode set and relay any data
+// for each port to the device attached to that port.
+void checkSerial()
+{
+  byte portId, serialData;
+  Stream* serialPort;
+
+  if (serialIndex > -1) {
+
+    // loop through all reporting (READ_CONTINUOUS) serial ports
+    for (byte i = 0; i < serialIndex + 1; i++) {
+      portId = reportSerial[i];
+      serialPort = getPortFromId(portId);
+      if (serialPort == NULL) {
+        continue;
+      }
+      // only the SoftwareSerial port that is "listening" can read data
+      if (portId > 7 && !((SoftwareSerial*)serialPort)->isListening()) {
```

Comment:
@rwaldron this appears to be the source of the error. Lines 218 - 220 should also be wrapped by:
```c++
#if defined(ARDUINO_ARCH_AVR)
...
#endif
```

Without those defines if you tried to compile on an ARM board like a Due or a Teensy, compilation would fail.